stglib/core/utils.py

```diff
@@ -1004,40 +1004,84 @@ def create_z(ds):
+                if ds.attrs["orientation"].upper() == "DOWN":
+                    depvar = (
+                        ds.attrs[name]
+                        - ds.attrs["initial_instrument_height"]
+                        + ds["bindist"].values
+                    )
+                elif ds.attrs["orientation"].upper() == "UP":
+                    depvar = (
+                        ds.attrs[name]
+                        - ds.attrs["initial_instrument_height"]
+                        - ds["bindist"].values
+                    )
+            else:
+                depvar = ds.attrs[name] - ds.attrs["initial_instrument_height"]
+            break
+
+    if "depvar" not in locals():
+        if "D_3" in ds:
+            if "bindist" in ds:
+                if ds.attrs["orientation"].upper() == "DOWN":
+                    depvar = np.nanmean(ds["D_3"]) + ds["bindist"].values
+                elif ds.attrs["orientation"].upper() == "UP":
+                    depvar = np.nanmean(ds["D_3"]) - ds["bindist"].values
+            else:
+                depvar = np.nanmean(ds["P_1ac"])
+            name = "sea_floor_depth_below_mean_sea_level"
+            longname = "depth below mean sea level from data"
+        elif "P_1ac" in ds:
+            if "bindist" in ds:
+                if ds.attrs["orientation"].upper() == "DOWN":
+                    depvar = np.nanmean(ds["P_1ac"]) + ds["bindist"].values
+                elif ds.attrs["orientation"].upper() == "UP":
+                    depvar = np.nanmean(ds["P_1ac"]) - ds["bindist"].values
+            else:
+                depvar = np.nanmean(ds["P_1ac"])
+            name = "sea_floor_depth_below_mean_sea_level"
+            longname = "depth below mean sea level from data"
         else:
-            presvar = ds.attrs["WATER_DEPTH"] - ds.attrs["initial_instrument_height"]
+            if "sea_floor_depth_below_sea_surface" in ds.attrs:
+                name = "sea_floor_depth_below_sea_surface"  # this is not ref to datum
+                longname = "depth below sea surface"
+            else:
+                name = "WATER_DEPTH"
+                longname = "depth below WATER_DEPTH ref"
+
+            if "bindist" in ds:
+                if ds.attrs["orientation"].upper() == "DOWN":
+                    depvar = (
+                        ds.attrs[name]
+                        - ds.attrs["initial_instrument_height"]
+                        + ds["bindist"].values
+                    )
+                elif ds.attrs["orientation"].upper() == "UP":
+                    depvar = (
+                        ds.attrs[name]
+                        - ds.attrs["initial_instrument_height"]
+                        - ds["bindist"].values
+                    )
+            else:
+                depvar = ds.attrs[name] - ds.attrs["initial_instrument_height"]
 
     if "bindist" in ds:
-        ds["depth"] = xr.DataArray(presvar, dims="depth")
+        ds["depth"] = xr.DataArray(depvar, dims="depth")
```

Comment:
At the end of all the if/else blocks above, is there a case where `depvar` will never be set? If so, we should raise an error indicating we are probably missing required attributes in the config file

Reply:
code will give KeyError if WATER_DEPTH is not in global attributes